backend/api/app/domains/user/repositories.py

```diff
@@ -23,6 +23,32 @@ def get_by_id(user_id):
     def get_all():
         return User.query.all()
     
+    @staticmethod
+    def updateUser(user):
+        try:
+            db.session.add(user)  
+            db.session.commit()  
+        except Exception as e:
+            db.session.rollback()  
+            raise e
+        
+    def update_password(user_id: uuid.UUID, hashed_password: str):
+        user = User.query.get(user_id)
+        if not user:
+            raise ValueError({"message": "User not found.", "status": 404})
+        user.password = hashed_password
+        db.session.commit()
+        
+    @staticmethod
+    def update_user(user: User):
+        db.session.commit()
+        
+    @staticmethod
+    def get_by_phone_number(phone_number: str):
+        return User.query.filter_by(phone_number=phone_number).first()
```

Comment:
Consider that you will possibly have a few accounts with the same number, but some will be deleted, and maybe they will not be the first to be returned and you will end with working with invalid data.